Source/WTF/wtf/Vector.h

```diff
@@ -56,48 +56,34 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(VectorBuffer);
 
 enum class NulloptBehavior : bool { Ignore, Abort };
 
-template <bool needsDestruction, typename T>
-struct VectorDestructor;
-
-template<typename T>
-struct VectorDestructor<false, T>
-{
-    static void destruct(T*, T*) { }
-};
-
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 
 template<typename T>
-struct VectorDestructor<true, T>
+struct VectorTypeOperations
 {
-    static void destruct(T* begin, T* end) 
+    static void destruct(T* begin, T* end)
     {
-        for (T* cur = begin; cur != end; ++cur)
-            cur->~T();
+        if constexpr (!std::is_trivially_destructible_v<T>) {
+            for (T* cur = begin; cur != end; ++cur)
+                cur->~T();
+        }
     }
-};
-
-template <bool needsInitialization, bool canInitializeWithMemset, typename T>
-struct VectorInitializer;
 
-template<bool canInitializeWithMemset, typename T>
-struct VectorInitializer<false, canInitializeWithMemset, T>
-{
-    static void initializeIfNonPOD(T*, T*) { }
-
-    static void initialize(T* begin, T* end)
+    static void destruct(std::span<T> span)
```

Comment:
I think callers can just use `std::ranges::destroy` instead.

Reply:
> It sounds a nice idea to use `std::ranges`. But, I don't want to do that in this change. Too risky.

Maybe we should off on this whole change. The entire thing is just "better style"; what is the primary benefit to the change? Does it fix any bugs?

Reply:
https://commits.webkit.org/276220@main introduced the performance regression for `WTF::Vector`.
I think one of the reason is TMP is too dificult. That's the motivation of this change.
https://bugs.webkit.org/show_bug.cgi?id=299008